components/fxa-client/android/src/main/java/mozilla/appservices/fxaclient/rust/LibFxAFFI.kt

```diff
@@ -56,7 +56,7 @@ internal interface LibFxAFFI : Library {
     fun fxa_get_access_token(fxa: FxaHandle, scope: String, ttl: Long, e: RustError.ByReference): RustBuffer.ByValue
     fun fxa_get_session_token(fxa: FxaHandle, e: RustError.ByReference): Pointer?
     fun fxa_get_current_device_id(fxa: FxaHandle, e: RustError.ByReference): Pointer?
-    fun fxa_authorize_auth_code(fxa: FxaHandle, clientId: String, scope: String, state: String, accessType: String, e: RustError.ByReference): Pointer?
+    fun fxa_authorize_auth_code(fxa: FxaHandle, len: Int, data: Pointer, e: RustError.ByReference): Pointer?
```

Comment:
nano-nit: I believe most C APIs that pass a `(data, length)` pair tend to pass them in that order, so it took me a few moments to understand what you were doing here when I saw the `len` parameter. I think changing this to be `params_data: Pointer, params_len: Int` would be a low-cost way to make it easier for future readers to parse this API correctly.

(An alternative would be to pass a `RustBuffer` struct here, but that's probably a more invasive change).